src/api/app/models/staging/staging_project.rb

```diff
@@ -1,10 +1,8 @@
 module Staging
   class StagingProject < Project
     has_many :staged_requests, class_name: 'BsRequest', foreign_key: :staging_project_id, dependent: :nullify
-    has_many :status_reports, through: :repositories, inverse_of: :checkable
-    has_many :checks, through: :status_reports
+    has_many :status_reports_for_repositories, source: :status_reports, through: :repositories, inverse_of: :checkable
```

Comment:
:thought_balloon: I don't really like the naming. I know `repo` vs. `arch` is what sets them apart technically but it's not what sets them apart in the workflow.

Reply:
These relationships shouldn't be public anyway - see https://github.com/coolo/open-build-service/commit/7e62af53e9e438ea32aeeec3b73c6cb616355e32

But I would prefer to do this as a followup